### PR TITLE
Add read_parquet

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
Depends on https://github.com/pandas-dev/pandas/pull/30842

File: fletcher/base.py [hunk 1]
```diff
@@ -637,6 +641,9 @@ def __getitem__(self, item):
         For a boolean mask, return an instance of ``ExtensionArray``, filtered
         to the values where ``item`` is True.
         """
+        if PANDAS_GE_0_26_0 and is_bool_dtype(item):
+            check_bool_array_indexer(self, item)
```
Comment (fjetter) [inline fletcher/base.py]: What it this doing?

Comment (jorisvandenbossche) [reply fletcher/base.py]: See https://github.com/pandas-dev/pandas/pull/30308. It is checking nullable boolean arrays for NAs (and raise if there are), and return a numpy array. 

So, actually, this needs to re-assign to `item`.

Comment (jorisvandenbossche) [inline fletcher/base.py]:
```suggestion
            item = check_bool_array_indexer(self, item)
```

File: fletcher/base.py [hunk 2]
```diff
@@ -1096,6 +1103,9 @@ def __getitem__(self, item):
...
```
Comment (jorisvandenbossche) [inline fletcher/base.py]:
```suggestion
            item = check_bool_array_indexer(self, item)
```

**Pull request merged**